app/Http/Controllers/Backend/SemesterController.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+namespace App\Http\Controllers\Backend;
+
+use App\Http\Controllers\Controller;
+use Illuminate\Http\Request;
+use App\Domains\Semester\Models\Semester;
+
+class SemesterController extends Controller
+{
+    /**
+     * Display a listing of the semesters.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        $semesters = Semester::all();
+        return view('backend.semesters.index', compact('semesters'));
+    }
+
+    /**
+     * Show the form for creating a new semester.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create()
+    {
+        return view('backend.semesters.create');
+    }
+
+    /**
+     * Store a newly created semester in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
+    {
+        $validatedData = $request->validate([
+            'title' => 'required|string|max:255',
+            'version' => 'required|integer',
+            'academic_program' => 'required|in:Undergraduate,Postgraduate',
```

Comment:
Shall we move academic program options into the Model level for easy update?

```
Rule::in(Semester::ACADEMIC_PROGRAMS)
```

where ACADEMIC_PROGRAMS is an array in Semester model.

Reply:
resolved it